sys/userlib/src/arch/arm_m.rs

```diff
@@ -65,11 +65,14 @@ pub unsafe extern "C" fn _start() -> ! {
                 @ a sym operand because it's a Rust func and may be mangled.
                 bl {main}
```

Comment:
What do you think about adding a userlib common entry point: `bl userlib_main` which does the following:
```
#[no_mangle]
pub fn userlib_main() {
    main();
    kipc::exit_current_task();
}
```

Then you won't have to repeat the same in different architectures. And have the logic commented in one place as well.

Reply:
That's a much better idea than all the arch-specific changes. I'll update the PR shortly.